bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/bridge/MatterBridge.java

```diff
@@ -0,0 +1,574 @@
+    private ConfigurationAdmin configAdmin;
+    private MatterBridgeSettings settings;
+
+    private final ItemRegistryChangeListener itemRegistryChangeListener;
+    private final RegistryChangeListener<Metadata> metadataRegistryChangeListener;
+    private final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
+    private boolean resetStorage = false;
+    private @Nullable ScheduledFuture<?> modifyFuture;
+    private @Nullable ScheduledFuture<?> reconnectFuture;
+    private RunningState runningState = RunningState.Stopped;
+
+    @Activate
+    public MatterBridge(final @Reference ItemRegistry itemRegistry, final @Reference MetadataRegistry metadataRegistry,
+            final @Reference MatterWebsocketService websocketService, final @Reference ConfigurationAdmin configAdmin) {
+        this.itemRegistry = itemRegistry;
+        this.metadataRegistry = metadataRegistry;
+        this.websocketService = websocketService;
+        this.configAdmin = configAdmin;
+        this.client = new MatterBridgeClient();
+        this.settings = new MatterBridgeSettings();
+
+        itemRegistryChangeListener = new ItemRegistryChangeListener() {
+            private boolean handleMetadataChange(Item item) {
+                if (metadataRegistry.get(new MetadataKey("matter", item.getUID())) != null) {
+                    updateModifyFuture();
+                    return true;
+                }
+                return false;
+            }
+
+            @Override
+            public void added(Item element) {
+                handleMetadataChange(element);
+            }
+
+            @Override
+            public void updated(Item oldElement, Item element) {
+                if (!handleMetadataChange(oldElement)) {
+                    handleMetadataChange(element);
+                }
+            }
+
+            @Override
+            public void allItemsChanged(Collection<String> oldItemNames) {
+                updateModifyFuture();
+            }
+
+            @Override
+            public void removed(Item element) {
+                handleMetadataChange(element);
+            }
+        };
+        this.itemRegistry.addRegistryChangeListener(itemRegistryChangeListener);
+
+        metadataRegistryChangeListener = new RegistryChangeListener<>() {
+            private void handleMetadataChange(Metadata element) {
+                if ("matter".equals(element.getUID().getNamespace())) {
+                    updateModifyFuture();
+                }
+            }
+
+            public void added(Metadata element) {
+                handleMetadataChange(element);
+            }
+
+            public void removed(Metadata element) {
+                handleMetadataChange(element);
+            }
+
+            public void updated(Metadata oldElement, Metadata element) {
+                handleMetadataChange(element);
```

Comment:
In itemRegistryChangeListener, you implemented a system to avoid calling `updateModifyFuture` twice if I correctly understand it.
Why not doing the same here ?